### PR TITLE
Make platforms optional

### LINKED ISSUE #2: tvOS and watchOS shouldn't mandatory
Great tool 👍 

I don't think all frameworks have to support the four platforms, in my case: I only need iOS and macOS, but in order to generate the xcodeproj, I have to set the four of them, if I don't it will simply crash. Is there a reason for this?

### PR DESCRIPTION
**Pull request by tbaranes**
Close #2 

This PRs allows the user to use only the platforms he's interested in: one, two, three or the four!
Once more, that's my first lines in ruby, so let me know if you prefer another syntax or way to implement this

### CODE REVIEW

File: lib/framework-generate/target.rb [hunk 1]
```diff
@@ -36,14 +36,31 @@ def target_build_settings(settings)
...
+      unless macos.nil?
+        settings['MACOSX_DEPLOYMENT_TARGET'] = FrameworkGenerate::Platform::deployment_target(macos)
+        settings['FRAMEWORK_SEARCH_PATHS[sdk=macosx*]'] = FrameworkGenerate::Platform::search_paths(macos)
+      end
+
+      ios = FrameworkGenerate::Platform::find_platform(@platforms, :ios)
+      unless ios.nil?
+        settings['IPHONEOS_DEPLOYMENT_TARGET'] = FrameworkGenerate::Platform::deployment_target(ios)
+        settings['FRAMEWORK_SEARCH_PATHS[sdk=iphone*]'] = FrameworkGenerate::Platform::search_paths(ios)
+      end
```
Comment (pmairoldi) [inline lib/framework-generate/target.rb:50]: `FrameworkGenerate::Platform::deployment_target` and `FrameworkGenerate::Platform::search_paths` take 2 parameters not one. Maybe you forgot to commit that change?

**Review by pmairoldi — CHANGES_REQUESTED** [review]
Aside from the runtime exception looks good. I think we can also change the `settings['TARGETED_DEVICE_FAMILY'] = '1,2,3,4'` to use the platforms specified in the target but that can be done separately.

File: lib/framework-generate/target.rb (hunk 1 above)
Comment (tbaranes) [reply lib/framework-generate/target.rb:50]: My bad, it's pushed

**Review by pmairoldi — APPROVED** [review]

**Comment by pmairoldi** [comment]
Published a new version (1.1.1) with these fixes. Enjoy!

**Pull request merged**